Docs/developer_guide/build_instructions/macos.md

```diff
@@ -5,74 +5,108 @@
+  - Additional configuration options to customize the application are described [here](overview.md#Custom_builds).
 
 ### General information
 
 Two projects are generated by either `cmake`, `ccmake` or `cmake-gui`. One of them is in the top-level bin directory `Slicer-SuperBuild` and the other one is in the subdirectory `Slicer-build`:
-- `Slicer-SuperBuild` manages all the external dependencies of Slicer (VTK, ITK, Python, ...). To build Slicer for the first time, run make (or build the solution file in Visual Studio) in `Slicer-SuperBuild`, which will update and build the external libraries and if successful will then build the subproject Slicer-build.
-- `Slicer-SuperBuild/Slicer-build` is the "traditional" build directory of Slicer.  After local changes in Slicer (or after an svn update on the source directory of Slicer), only running make (or building the solution file in Visual Studio) in `Slicer-SuperBuild/Slicer-build` is necessary (the external libraries are considered built and up to date).
 
-*Warning:* An significant amount of disk space is required to compile Slicer in Debug mode (>10GB)
+- `Slicer-SuperBuild` manages all the external dependencies of Slicer (VTK, ITK, Python, ...). To build Slicer for the first time, run `make` in `Slicer-SuperBuild`, which will update and build the external libraries and if successful will then build the subproject Slicer-build.
+- `Slicer-SuperBuild/Slicer-build` is the "traditional" build directory of Slicer.  After local changes in Slicer (or after an git update on the source directory of Slicer), only running `make` in `Slicer-SuperBuild/Slicer-build` is necessary (the external libraries are considered built and up to date).
+
+*Warning:* An significant amount of disk space is required to compile Slicer in Debug mode (>20GB)
 
 *Warning:* Some firewalls will block the git protocol. See more information and solution [here](../overview.html#firewall-is-blocking-git-protocol).
 
 ## Build Slicer
 
 After configuration, start the build process in the `Slicer-SuperBuild` directory
 
-- Start a terminal and type the following (you can replace 4 by the number of processor cores in the computer):
-```
+- Start a terminal and type the following (you can replace 4 by the number of processor cores in the computer. You can find out the number of available cores by running `sysctl -n hw.ncpu`):
+
+```bash
 cd ~/Projects/Slicer-SuperBuild
 make -j4
 ```
 
-When using the -j option, the build will continue past the source of the first error. If the build fails and you don't see what failed, rebuild without the -j option. Or, to speed up this process build first with the -j and -k options and then run plain make. The -k option will make the build keep going so that any code that can be compiled independent of the error will be completed and the second make will reach the error condition more efficiently.
+### Debugging the build process
```

Comment:
Should we move this onto its own page ? So that it can linked from Linux and macOS build instructions?

Reply:
Maybe common errors and debugging can be on the same page?

Reply:
Please keep linux and mac instructions on completely separate pages. We should only put very special things on the "Common errors" page: that are common to all operating systems _and_ not part of the build instructions (exceptions). But we may even consider removing that section (depending on how its content evolves), because it is much easier to find information if everything is on a single page.

Trying to reduce duplication in build instructions leads really bad results, as we learned in the past. It is obviously bad for developers (I had to keep scrolling the documentation pages really carefully and figure out for each section if it is applicable to my platform). But it is also bad for maintainers and results in lower quality documentation, because when you update build instructions for linux or mac then you either end up just writing instructions without testing on the other platform; or you build and edit instructions on linux and mac at the same time and make mistakes because you are too distracted.